File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i = 0
        e = len(height)-1
        maxArea = 0
        while i < e:
            area = (e - i) * min(height[i], height[e])
            maxArea = max(area,maxArea)
            if height[i]<height[e]:
                i += 1
            else:
                e -= 1
        return maxArea

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0, r = heights.length - 1;

        int vol = Integer.MIN_VALUE;
        while (l < r) {
            int newVol = Math.min(heights[l], heights[r]) * (r - l);
            vol = Math.max(newVol, vol);

            if (heights[l] < heights[r]) l++;
            else r--;
        }

        return vol;
    }
}
